spark/src/main/scala/org/apache/spark/sql/delta/ResolveDeltaPathTable.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.analysis.UnresolvedTable
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.connector.catalog.CatalogV2Implicits.MultipartIdentifierHelper
+import org.apache.spark.sql.delta.catalog.DeltaTableV2
+import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation
+
+/**
+ * Replaces [[UnresolvedTable]]s if the plan is for direct query on files.
+ */
+case class ResolveDeltaPathTable(sparkSession: SparkSession) extends Rule[LogicalPlan] {
+
+  private def maybeSQLFile(u: UnresolvedTable): Boolean = {
+    sparkSession.sessionState.conf.runSQLonFile && u.multipartIdentifier.size == 2
```

Comment:
This seems to copy ideas from [ResolveSQLOnFile](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala#L50) from spark? Is there a reason we can't leverage that here, and let `DeltaDataSource.getTable` produce the `DeltaTableV2` we need?

Reply:
Ugh, `UnresolvedTable` != `UnresolvedRelation`, and `it looks like the data source code uses `UnresolvedRelation` while `UnresolvedPathBasedDeltaTable` uses`UnresolvedTable`.

Reply:
1. in the parser, this PR uses `UnresolvedDeltaPathOrIdentifier` and it will produce `UnresolvedTable` on table identifiers (including the file path table delta.`path`)
2. If we create `UnresolvedRelation` as the child of `VacuumTableCommand`, the resolved relation from Apache Spark will be a Parquet data source relation. There is some issue with my debugger and I haven't figured out the reason.
3. In the analyzer rule `ResolveRelations`, both `UnresolvedTable` and `UnresolvedRelation` are processed. `UnresolvedTable` always result in `ResolvedTable`, while `UnresolvedRelation` results in `SubqueryAlias` with various nodes. I think using `UnresolvedTable` is simpler here. Any reason why we should use `UnresolvedRelation`?

Reply:
>  I think using `UnresolvedTable` is simpler here. Any reason why we should use `UnresolvedRelation`?

Yeah, `UnresolvedRelation` only makes sense if it allows us to reuse existing machinery in some way. But:

> resolved relation from Apache Spark will be a Parquet data source relation

That's... awkward. Tho I've noticed that the file index for Delta _is_ parquet source because that's the physical file format Delta reads. Is there no trace of Delta in the resulting scan node, tho?